src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/CaCaseloadService.kt

```diff
@@ -0,0 +1,492 @@
+    GroupedByCom(
+      withStaffCode = emptyList(),
+      withStaffUsername = emptyList(),
+      withNoComId = emptyList(),
+    ),
+  ) { acc, caCase ->
+    if (caCase.probationPractitioner?.staffCode != null) {
+      acc.withStaffCode += caCase
+    } else if (caCase.probationPractitioner?.staffUsername != null) {
+      acc.withStaffUsername += caCase
+    } else {
+      acc.withNoComId += caCase
+    }
+    return@fold acc
+  }
+
+  private fun createNotStartedLicenceForCase(cases: List<ManagedCase>): List<CaCase> = cases.map { c ->
+
+    // Default status (if not overridden below) will show the case as clickable on case lists
+    var licenceStatus = NOT_STARTED
+
+    if (isBreachOfTopUpSupervision(c)) {
+      // Imprisonment status indicates a breach of top up supervision order - not clickable (yet)
+      licenceStatus = OOS_BOTUS
+    } else if (isRecall(c)) {
+      // Offender is subject to an active recall - not clickable
+      licenceStatus = OOS_RECALL
+    } else if (c.cvlFields.isInHardStopPeriod) {
+      licenceStatus = TIMED_OUT
+    }
+
+    val com = c.deliusRecord?.managedOffenderCrn?.staff
+
+    val releaseDate = c.nomisRecord?.confirmedReleaseDate ?: c.nomisRecord?.conditionalReleaseDate
+
+    CaCase(
+      name = "${c.nomisRecord?.firstName} ${c.nomisRecord?.lastName}",
+      prisonerNumber = c.nomisRecord?.prisonerNumber!!,
+      releaseDate = releaseDate!!,
+      releaseDateLabel = if (c.nomisRecord.confirmedReleaseDate != null) {
+        "Confirmed release date"
+      } else {
+        "CRD"
+      },
+      licenceStatus = licenceStatus,
+      nomisLegalStatus = c.nomisRecord.legalStatus,
+      isDueForEarlyRelease = c.cvlFields.isDueForEarlyRelease,
+      isInHardStopPeriod = c.cvlFields.isInHardStopPeriod,
+      tabType = determineCaViewCasesTab(c.nomisRecord, c.cvlFields, licence = null),
+      probationPractitioner = ProbationPractitioner(
+        staffCode = com?.code,
+        name = com?.let { "${it.forenames} ${it.surname}" },
+      ),
+    )
+  }
+
+  private fun filterOffendersEligibleForLicence(offenders: List<CaseloadItem>): List<CaseloadItem> {
+    val eligibleOffenders = offenders
+      .asSequence()
+      .filter { offender -> !isParoleEligible(offender.prisoner.paroleEligibilityDate, clock) }
+      .filter { offender -> offender.prisoner.legalStatus != "DEAD" }
+      .filter { offender -> !(offender.prisoner.indeterminateSentence ?: false) }
+      .filter { offender -> offender.prisoner.conditionalReleaseDate != null }
+      .filter { offender ->
+        val status = offender.prisoner.status
+        return@filter (status != null && (status.startsWith("ACTIVE") || status == "INACTIVE TRN"))
+      }
+      .filter { offender ->
+        return@filter (
+          offender.prisoner.confirmedReleaseDate
+            ?: offender.prisoner.conditionalReleaseDate
+          )?.isAfter(LocalDate.now(clock)) == true
```

Comment:
I don't think `isAfter` is appropriate here, unless Kotlin's implementation includes `today` in `isAfter` (I assume it doesn't)
We still need to return these cases if the release date is today

Reply:
Now this was removed and using isEligibleForCvl from eligibility service